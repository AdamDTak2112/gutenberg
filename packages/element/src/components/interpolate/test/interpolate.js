/**
 * External dependencies
 */
import TestRenderer from 'react-test-renderer';

/**
 * Internal dependencies
 */
import Interpolate from '../interpolate';

describe( 'Interpolate', () => {
	it( 'renders expected result from component', () => {
		const TestComponent = () => {
			const image = <img alt={ 'An test' } src={ '/file' } />;
			const url = 'https://example.org';
			return <Interpolate
				extra={ 'and fries' }
				foo={ 'bar' }
				image={ image }
			>
				This is an awesome cheeseburger %%extra%%! Go <a href={ url }>here</a> to eat at the %%foo%%.
			</Interpolate>;
		};
		const tree = TestRenderer
			.create( <TestComponent /> )
			.toJSON();
		expect( tree ).toMatchSnapshot();
	} );
} );
