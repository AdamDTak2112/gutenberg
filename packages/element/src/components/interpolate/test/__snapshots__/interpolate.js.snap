// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Interpolate renders expected result from component 1`] = `
Array [
  "This is an awesome cheeseburger ",
  "and fries",
  "! Go ",
  <a
    href="https://example.org"
  >
    here
  </a>,
  " to eat at the ",
  "bar",
  ".",
]
`;
